src/ext/oblivc/obliv_transport.c

```diff
@@ -0,0 +1,343 @@
+#include <obliv_common.h>
+#include <obliv_bits.h>
+#include <obliv_yao.h>
+#include <commitReveal.h>
+#include <nnob.h>
+#include <assert.h>
+#include <errno.h>      // libgcrypt needs ENOMEM definition
+#include <inttypes.h>
+#include <stdio.h>      // for protoUseStdio()
+#include <string.h>
+#include <netinet/ip.h>
+#include <netinet/tcp.h>
+#include <netdb.h>
+#include <pthread.h>
+#include <unistd.h>
+#include <gcrypt.h>
+
+// --------------------------- Transports -----------------------------------
+struct stdioTransport
+{ ProtocolTransport cb;
+  bool needFlush;
+};
+
+// Ignores 'dest' parameter. So you can't osend to yourself
+static bool* stdioFlushFlag(ProtocolTransport* pt) 
+  { return &((struct stdioTransport*)pt)->needFlush; }
+
+static int stdioSend(ProtocolTransport* pt,int dest,const void* s,size_t n)
+{ *stdioFlushFlag(pt)=true;
+  return fwrite(s,1,n,stdout); 
+}
+
+static int stdioRecv(ProtocolTransport* pt,int src,void* s,size_t n)
+{ 
+  bool *p = stdioFlushFlag(pt);
+  if(*p) { fflush(stdout); *p=false; }
+  return fread(s,1,n,stdin); 
+}
+
+static void stdioCleanup(ProtocolTransport* pt) {}
+
+// Extremely simple, no multiplexing: two parties, one connection
+static struct stdioTransport stdioTransport 
+  = {{2, NULL, stdioSend, stdioRecv, stdioCleanup},false};
+
+void protocolUseStdio(ProtocolDesc* pd)
+  { pd->trans = &stdioTransport.cb; }
+
```

Comment:
This PR seems to overwrite a whole bunch of commits I made, which changed the way network profiling worked (specifically, PROFILE_NETWORK was removed, which is why I noticed). I suspect other commits by myself and others are also overwritten, though I haven't specifically looked for obsolete code. Github does not detect the conflict for some reason - possibly because of the fact that you split everything into separate files (which is a very good thing, otherwise).